File: app/src/main/java/game/engine/MyGameEngine.java
package game.engine;

import adapter.Adapter;
import edu.austral.dissis.chess.gui.*;
import factory.BoardFactory;
import game.managers.*;
import game.movement.Movement;
import game.status.GameStatus;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import static factory.PieceFactory.createQueen;

public class MyGameEngine implements GameEngine {

    private final GameStatus status = new GameStatus(new ClassicTurnManager(), BoardFactory.createClassicBoard());
    private final List<GameManager> managers;


    //classic game engine
    public MyGameEngine() {
        List<GameManager> managers = new ArrayList<>();
        managers.add(new MoveGameManager());
        managers.add(new PromotionManager(createQueen(PlayerColor.WHITE)));
        managers.add(new ClassicWinManager());
        this.managers = managers;
    }

    public MyGameEngine(List<GameManager> managers) {
        this.managers = managers;
    }

    @NotNull
    @Override
    public MoveResult applyMove(@NotNull Move move) {
        Movement movement = Adapter.movement(move);
        MoveResult moveResult = new InvalidMove("Error");
        for (GameManager manager : managers) {
            MoveResult moveResult1 = manager.evaluateMovement(movement, status);
            if (moveResult1 == null) continue;
            else moveResult = moveResult1;
        }
        return moveResult;
    }

    @NotNull
    @Override
    public InitialState init() {
        return new InitialState(Adapter.boardSize(status.board().columns(),status.board().rows()),Adapter.pieces(status.board()), PlayerColor.WHITE);
    }
}

File: app/src/main/kotlin/edu/austral/dissis/chess/MyGameEngine.kt
package edu.austral.dissis.chess

import adapter.GameAdapter
import edu.austral.dissis.chess.gui.*
import factory.GameFactory
import interfaces.Game

class MyGameEngine : GameEngine {

    private val game: Game = GameFactory.classicGame();
    private val adapter: GameAdapter = GameAdapter(game);

    override fun applyMove(move: Move): MoveResult {

        TODO("Not implemented yet")
    }

    override fun init(): InitialState {
        val pieces = adapter.getPieces()
        return InitialState(BoardSize(8, 8), pieces, PlayerColor.WHITE);
    }
}
